$OpenBSD: patch-src_ui_theme_c,v 1.1 2010/05/11 20:49:39 jasper Exp $

Plug a memory leak in theme optimisation.
Upstream git bc14af3aaa0cd4c51d4558209adf2fee9878b071

--- src/ui/theme.c.orig	Tue May 11 22:43:55 2010
+++ src/ui/theme.c	Tue May 11 22:44:37 2010
@@ -2559,11 +2559,13 @@ meta_theme_replace_constants (MetaTheme   *theme,
         {
           if (meta_theme_lookup_int_constant (theme, t->d.v.name, &ival))
             {
+              g_free (t->d.v.name);
               t->type = POS_TOKEN_INT;
               t->d.i.val = ival;
             }
           else if (meta_theme_lookup_float_constant (theme, t->d.v.name, &dval))
             {
+              g_free (t->d.v.name);
               t->type = POS_TOKEN_DOUBLE;
               t->d.d.val = dval;
             }
